pkg/v2/mcapi.go

```diff
@@ -122,39 +122,88 @@ func (client *Client) InitSystemInfo() error {
 }
 
 // CreateCommonStatus : create a common API status object based on the OpenAPI client response
-func CreateCommonStatus(response *client.StatusObject) *common.ResponseStatus {
+func CreateCommonStatus(logger logr.Logger, response *[]client.StatusResourceInner) *common.ResponseStatus {
 
 	status := common.ResponseStatus{}
 
+	// Track if this status contains one or more non-info response types
+	found := false
+
 	if response != nil {
-		status.ResponseType = response.Status[0].GetResponseType()
-		status.ResponseTypeNumeric = int(response.Status[0].GetResponseTypeNumeric())
-		status.Response = response.Status[0].GetResponse()
-		status.ReturnCode = int(response.Status[0].GetReturnCode())
-		status.Time = time.Unix(int64(response.Status[0].GetTimeStampNumeric()), 0)
+
+		// The status response may contain multiple entries, so skip over an "Info" types
+		for _, s := range *response {
+
+			if *s.ResponseType == "Info" {
+				logger.V(2).Info("create common status (info)",
+					"ResponseType", *s.ResponseType,
+					"ResponseTypeNumeric", *s.ResponseTypeNumeric,
+					"Response", *s.Response,
+				)
+			} else {
+				if found {
+					// print warning since there are multiple non-info messages
+					logger.V(0).Info("create common status multiple non-info responses",
+						"ResponseType", *s.ResponseType,
+						"ResponseTypeNumeric", *s.ResponseTypeNumeric,
+						"Response", *s.Response,
+						"ReturnCode", *s.ReturnCode,
+					)
+				}
+				logger.V(2).Info("create common status",
+					"ResponseType", *s.ResponseType,
+					"ResponseTypeNumeric", *s.ResponseTypeNumeric,
+					"Response", *s.Response,
+				)
+				found = true
+				status.ResponseType = s.GetResponseType()
+				status.ResponseTypeNumeric = int(s.GetResponseTypeNumeric())
+				status.Response = s.GetResponse()
+				status.ReturnCode = int(s.GetReturnCode())
+				status.Time = time.Unix(int64(s.GetTimeStampNumeric()), 0)
+			}
+		}
 	}
 
 	return &status
 }
 
 // CreateCommonStatusFromStatus : create a common API status object based on the OpenAPI client response
-func CreateCommonStatusFromStatus(logger logr.Logger, response *openapiclient.StatusResourceInner) *common.ResponseStatus {
+func CreateCommonStatusFromStatus(logger logr.Logger, response *[]openapiclient.StatusResourceInner) *common.ResponseStatus {
```

Comment:
Is there someway to avoid this code repetition?  If not, maybe explain in the comments to help ensure that someone doesn't update one function without making the same change in the other?

Reply:
I missed that after the refactor, but have removed the redundant function now. https://github.com/Seagate/seagate-exos-x-api-go/pull/43/commits/585d5fdb7ab6884238a8e2bb183bd4647d5fe122